ledger-core/virtual/integration/deactivate_test.go

```diff
@@ -139,3 +149,254 @@ func TestVirtual_DeactivateObject(t *testing.T) {
 		})
 	}
 }
+
+func TestVirtual_CallMethod_On_CompletelyDeactivatedObject(t *testing.T) {
+	insrail.LogCase(t, "C4975")
+	stateTestCases := []struct {
+		name        string
+		objectState contract.StateFlag
+	}{
+		{
+			name:        "call on validated state",
+			objectState: contract.CallValidated,
+		},
+		{
+			name:        "call on dirty state",
+			objectState: contract.CallDirty,
+		},
+	}
+
+	for _, stateTest := range stateTestCases {
+		t.Run(stateTest.name, func(t *testing.T) {
+
+			callTypeTestCases := []struct {
+				name     string
+				callType payload.CallTypeNew
+				errorMsg string
+			}{
+				{
+					name:     "call method",
+					callType: payload.CTMethod,
+					errorMsg: "attempt to call method on object that is completely deactivated",
+				},
+				{
+					name:     "call constructor",
+					callType: payload.CTConstructor,
+					errorMsg: "attempt to construct object that was completely deactivated",
+				},
+			}
+
+			for _, callTypeTest := range callTypeTestCases {
+				t.Run(callTypeTest.name, func(t *testing.T) {
+					defer commontestutils.LeakTester(t)
+
+					mc := minimock.NewController(t)
+
+					server, ctx := utils.NewUninitializedServer(nil, t)
+					defer server.Stop()
+
+					runnerMock := logicless.NewServiceMock(ctx, t, func(execution execution.Context) string {
+						return execution.Request.CallSiteMethod
+					})
+					server.ReplaceRunner(runnerMock)
+
+					server.Init(ctx)
+					server.IncrementPulseAndWaitIdle(ctx)
+
+					var (
+						object    = reference.NewSelf(server.RandomLocalWithPulse())
+						prevPulse = server.GetPulse().PulseNumber
+					)
+
+					server.IncrementPulseAndWaitIdle(ctx)
+					Method_PrepareObject(ctx, server, payload.Inactive, object, prevPulse)
+
+					gotResult := make(chan struct{})
+
+					typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+					typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+
+						assert.Equal(t, res.Callee, object)
+						contractErr, sysErr := foundation.UnmarshalMethodResult(res.ReturnArguments)
+						require.Equal(t, &foundation.Error{callTypeTest.errorMsg}, contractErr)
+						require.NoError(t, sysErr)
+
+						gotResult <- struct{}{}
+
+						return false // no resend msg
+					})
+
+					pl := payload.VCallRequest{
+						CallType:            callTypeTest.callType,
+						CallFlags:           payload.BuildCallFlags(contract.CallIntolerable, stateTest.objectState),
+						Caller:              server.GlobalCaller(),
+						Callee:              object,
+						CallSiteDeclaration: testwallet.GetClass(),
+						CallSiteMethod:      "MyFavorMethod",
+						CallOutgoing:        reference.NewSelf(object.GetLocal()),
```

Comment:
use server.BuildRandomOutgoingWithPulse() for outgoing

Reply:
done